pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/util/SystemExecutables.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.gradle.python.util;
+
+import org.apache.tools.ant.taskdefs.condition.Os;
+
+/**
+ * Enum that allows for platform independent resolution of executables
+ */
+public enum SystemExecutables {
```

Comment:
I like the abstraction, but I don't know if this is the right abstraction. I'm wondering if a `CrossPlatformExecutable` interface and we would have a PythonExecutable and ActivateExecutable that would implement it, then the python one can have a method `withVersion(String version)`.